src/Examine/Logging/ILoggingService.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Examine.Logging
+{
+    public interface ILoggingService
+    {
+        void Log(LogEntry logEntry);
```

Comment:
I'd recommend using the Microsoft logging abstractions rather than introducing custom logging.

Reply:
@nzdev we are low level, which means we can't be dependand on external loggings, so I build abstraction where you can inject what ever you are using, in my case it is serilog :), in you microsoft logging :), that is just plain simple abstraction

Reply:
Also we don't want introduce any additional nuget packages as that is hell when you getting one small repo and getting huge abstraction package from microsoft 👎

Reply:
Microsoft logging is the abstraction and its supported by many libraries including serilog etc. I don't think it's appropriate to make up another abstraction when one exists that's well supported in the .net ecosystem. It's appropriate to depend on other packages even for libraries, this library is not low level as it's abstracting and adapting license and azure storage.

Reply:
I like the idea in principle of less dependences, however I think it's advantageous to the project to take on microsoft logging abstractions as a dependency.

Advantages:
- Well documented
- 600+ million downloads on nuget.
- Well tested. Battle ready from use in millions of projects
- Familiar to .net developers, easier for newcomers to get started
- Adapters written for all major .net logging libraries. No need to code, register, test, maintain adapters per logging library.
- Other logging abstractions such as common.logging and liblog have fallen out of favour/ support.

Disadvantages:
- One extra nuget package to download.

I think on the balance, microsoft logging abstractions are the way to go.

Reply:
@nzdev will look into that and make decision today later, just needed quick abstraction to test on our project, so can refactor that at anypoint :)

Reply:
This PR is becoming insane 😅 let's keep to to an absolute minimum. Please do not add logging, add todos or something and make it another task. I would prefer zero more dependencies. This also had to work with umbraco "now" which has no idea about MS logging. If we implement it, it will be our own abstraction or an event until ported to netcore.

Reply:
> This PR is becoming insane 😅 let's keep to to an absolute minimum. Please do not add logging, add todos or something and make it another task. I would prefer zero more dependencies. This also had to work with umbraco "now" which has no idea about MS logging. If we implement it, it will be our own abstraction or an event until ported to netcore.
I added really basic logging there with own abstraction to be able track better issues, which can help definitely support in future, so I would leave that own abstraction there, not sure if you agree with me, but tracelogs are really hard to track, so if someone is using that outside of umbraco we have option to asked them to use anyother logging :)

Reply:
MS logging can wait till the netstandard version so sure, the simple abstraction for now is fine.

Reply:
so I am sticking with my small abstraction which by default will behave as Trace write, so we don't introduce anychanges in code which was already there, so we don't really need test that. In future we should do file logging, but not going to be part of that PR :)